packages/rpc-core/src/rpc-methods/__typetests__/get-block-type-test.ts

```diff
@@ -0,0 +1,1256 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { Blockhash, TransactionVersion } from '@solana/transactions';
+import assert from 'assert';
+
+import { TransactionError } from '../../transaction-error';
+import {
+    Base58EncodedBytes,
+    Base58EncodedDataResponse,
+    Base64EncodedDataResponse,
+    LamportsUnsafeBeyond2Pow53Minus1,
+    Reward,
+    TokenBalance,
+    TransactionStatus,
+    U64UnsafeBeyond2Pow53Minus1,
+} from '../common';
+import { mockTypeTestResponse, mockTypeTestRpc } from './common';
+
+const rpc = mockTypeTestRpc();
+
+function assertBase(
+    response: {
+        blockhash: string;
+        blockHeight: bigint;
+        blockTime: number;
+        parentSlot: bigint;
+        previousBlockhash: string;
+    } & { [key: string]: unknown }
+) {
+    response.blockhash satisfies string;
+    response.blockHeight satisfies bigint;
+    response.blockTime satisfies number;
+    response.parentSlot satisfies bigint;
+    response.previousBlockhash satisfies string;
+}
+
+// First overload
+// Rewards set to `false`
+{
+    const response = mockTypeTestResponse(
+        rpc.getBlock(0n, {
+            // No extra configs
+            rewards: false,
+            transactionDetails: 'none',
+        })
+    );
```

Comment:
What if, instead of this extractor, we just used the API directly?
```suggestion
    const response = await rpc
      .getBlock(0n, {
          // No extra configs
          rewards: false,
          transactionDetails: 'none',
      })
      .send();
```

The type of `response` will be correct in that case, _and_ this test will still catch breakages with the implementation/typing of the `send()` method.

Reply:
You can't await in the file root, so you have to set up a function for this and it looks super ugly. That was the reason I went through the trouble

Reply:
It does say we can set certain module options though, maybe for this directory only?

Reply:
> You can't await in the file root, so you have to set up a function for this and it looks super ugly…

Oh yeah, fuck that; just wrap the whole entire test in:

```ts
(async () => {
  // Every test.
})();
```